src/util/execute-lifecycle-script.js

```diff
@@ -116,8 +121,22 @@ export async function executeLifecycleScript(
   // split up the path
   const pathParts = (env[constants.ENV_PATH_KEY] || '').split(path.delimiter);
 
-  // add node-gyp
-  pathParts.unshift(path.join(__dirname, '..', '..', 'bin', 'node-gyp-bin'));
+  // Include node-gyp version that was bundled with the current Node.js version,
+  // if available.
+  pathParts.unshift(
+    path.join(
+      path.dirname(process.execPath), 'node_modules', 'npm', 'bin', 'node-gyp-bin',
+    ),
+  );
+  pathParts.unshift(
+    path.join(
```

Comment:
this stuff can use `path.resolve` which gets rid of the `..` from the resulting string, and you can pass a single arg with `/`. Yes, that works on windows.

```sh-session
$ node -pe "require('path').win32.resolve('/root', 'some/path/../slashes')"
\root\some\slashes
```